semgrep-core/src/core/ast/Ugly_print_AST.ml

```diff
@@ -197,41 +197,65 @@ class base_printer : printer_t =
       if self#needs_parens any then combine [ b "("; res; b ")" ] else res
   end
 
-class python_printer : printer_t =
+(* Printing for constructs that are identical across many languages. Unlike
+ * base_printer, it's not expected that every printer inherit from this. But
+ * many printers should. This can be split up as needed so that printers can use
+ * pieces a la carte, but care should be taken to avoid making the inheritance
+ * graph overly complicated.
+ *
+ * This is virtual rather than a subclass of base_printer so that language
+ * printers can pull pieces from this and other classes as needed to concisely
+ * build up a complete printer. *)
+class virtual common_printer =
```

Comment:
maybe at some point we can rename common_printer to c_like_printer, or classic_parencalls_printer,
or something more descriptive than common.

Reply:
Yes! I intend to do this. I tried to come up with a more descriptive name right now, but I wasn't sure what this would turn in to as we added more printing and more languages. I have a feeling that the boundaries will become clearer as we add printing.